### PR TITLE
Allow for using the system time format in the timeline view

### PR DESCRIPTION
**Pull request by Rich86man**

### CODE REVIEW

**Comment by Rich86man** [comment]
System time format in the timeline view
Calendar density (verticalDiff)

**Comment by richardtop** [comment]
@Rich86man thanks for the pull request, I've found a couple of errors, before it could be merged.
At first, if the time format is set to `system`, the calendar does not respect time settings and always 
sets timeline to `12H` format. 
Here is the screenshot, note, that even if the format is set to `24H`, the auxiliary views on timeline still use `12H` format. I suppose, to fix that, you need to add condition on how to format the time in `timeStringsSystem()` and use either `24H` or `12H` format. 
img_3797

File: Source/CalendarStyle.swift [hunk 1]
```diff
@@ -42,10 +42,23 @@ public class TimelineStyle {
...
   public init() {}
 }
 
 public class CurrentTimeIndicatorStyle {
   public var color = UIColor.red
   public init() {}
 }
+
+public enum TimelineTimeFormat {
+  case twentyFourHour, twelveHour, system
```
Comment (richardtop) [inline Source/CalendarStyle.swift]: I was going to decouple `CalendarConfig` to a separate class, similar to `CalendarStyle`. The concept is the same, the only difference that `CalendarStyle` is responsible for defining looks only, and `CalendarConfig` is about time/date/calendar settings in general.

Comment (Rich86man) [inline Source/CalendarStyle.swift]: Thats probably a good idea. The time format didn't really feel like a "style"

**Comment by Rich86man** [comment]
The current time indicator was not configurable, so I've added that to its style. I agree that we should move to a CalendarConfig, but until we flesh out how that works, the style works for now. 

I wasn't able to reproduce the issue where setting the format to `24H` caused the timeline to show up in `12H` format. I've updated the dark style in the example project to use `24H` format and it shows up in the `24H` project.

**Comment by richardtop** [comment]
@Rich86man looks good, but there is a difference, how 24/12h format is represented in the timeline. Please, refer to `Generator.swift` and update the format accordingly. For example, in `12h` format there are no zeros to indicate minutes, but only `am/pm` sign in lowercase.
image

Also, in `Generator.swift` `AM/PM` are indicated in capital letters, as in older versions of iOS. This should be updated accordingly, but you don't have to do it in this pull request.

**Comment by gavinpquinn** [comment]
Any chance this is going to go in? Seems a bit old :)

**Comment by richardtop** [comment]
Closing this pull request, as the same feature was implemented in #86.

**Pull request closed without merging**